astroquery/mast/cloud.py

```diff
@@ -0,0 +1,191 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+"""
+Cloud Access
+============
+
+This file contains functionality for accessing MAST holdings in the cloud.
+"""
+
+import os
+import warnings
+import threading
+
+from astropy.logger import log
+from astropy.utils.console import ProgressBarOrSpinner
+
+from ..exceptions import NoResultsWarning
+
+from . import fpl
+
+
+class CloudAccess(object):  # pragma:no-cover
+    """
+    Class encapsulating access to MAST data in the cloud.
+    """
+
+    def __init__(self, provider="AWS", profile=None, verbose=False):
+        """
+        Initialize class te enable downloading public files from S3 
+        instead of STScI servers.
+        Requires the boto3 and botocore libraries to function.
+
+        Parameters
+        ----------
+        provider : str
+            Which cloud data provider to use. Currently only S3 is supported,
+            so at the moment this argument is ignored.
+        profile : str
+            Profile to use to identify yourself to the cloud provider (usually in ~/.aws/config).
+        verbose : bool
+            Default False. Display extra info and warnings if true.
+        """
+
+        import boto3
+        import botocore
+
+        if profile is not None:
+            self.boto3 = boto3.Session(profile_name=profile)
+        else:
+            self.boto3 = boto3
+        self.botocore = botocore
+
+        self.pubdata_bucket = "stpubdata"
+
+        if verbose:
+            log.info("Using the S3 STScI public dataset")
+            log.warning("Your AWS account will be charged for access to the S3 bucket")
+            log.info("See Request Pricing in https://aws.amazon.com/s3/pricing/ for details")
+            log.info("If you have not configured boto3, follow the instructions here: "
+                     "https://boto3.readthedocs.io/en/latest/guide/configuration.html")
+
+    def get_uri(self, data_product, include_bucket=True, full_url=False):
```

Comment:
Since dataURI is a thing in the data_product, I find it potentially confusing for this to be called `get_uri()`.  Maybe something like `get_cloud_uri()` would be a little better?

Reply:
Haha, it used to be `get_cloud_uri` but I decided that was redundant when I moved it into the cloud class. Was not even thinking that the input is also a URI. 😛